homeassistant/components/sensor/hue.py

```diff
@@ -0,0 +1,513 @@
+        return UOM_HUMIDITY
+
+
+class CLIPLightLevel(CLIPSensor):
+    def __init__(self, sensor, request_bridge_update, bridge):
+        """Initialize the sensor."""
+        super().__init__(sensor, request_bridge_update, bridge)
+
+    @property
+    def state(self):
+        """Return the state of the CLIP sensor."""
+        return self.sensor.lightlevel
+
+    @property
+    def device_class(self):
+        """Return the device class of the CLIP sensor."""
+        return DEVICE_CLASS_ILLUMINANCE
+
+    @property
+    def unit_of_measurement(self):
+        """Return the uom of the CLIP sensor."""
+        return UOM_ILLUMINANCE
+
+
+class CLIPSwitch(CLIPSensor):
+    """Class representing a CLIP Switch."""
+    def __init__(self, sensor, request_bridge_update, bridge):
+        super().__init__(sensor, request_bridge_update, bridge)
+        self._last_update_time = self.sensor.lastupdated
+
+    @property
+    def state(self):
+        """Return the switch's buttonevent if the lastupdated variable changed."""
+        current_time = self.sensor.lastupdated
+        if current_time == self._last_update_time:
+            return None
+        else:
+            self._last_update_time = current_time
+            return self.sensor.buttonevent
+
+    @property
+    def icon(self):
+        """Return the remote icon."""
+        return ICON_REMOTE
+
+
+class CLIPTemperature(CLIPSensor):
+    def __init__(self, sensor, request_bridge_update, bridge):
+        """Initialize the sensor."""
+        super().__init__(sensor, request_bridge_update, bridge)
+
+    @property
+    def state(self):
+        """Return the state of the CLIP sensor."""
+        return round(self.sensor.temperature / 100, 1)
+
+    @property
+    def device_class(self):
+        """Return the device class of the CLIP sensor."""
+        return DEVICE_CLASS_TEMPERATURE
+
+    @property
+    def unit_of_measurement(self):
+        """Return the uom of the CLIP sensor."""
+        return TEMP_CELSIUS
+
+
+class ZGPSwitch(Entity):
+    """Class representing a ZGP (Hue Tap) Switch."""
+    def __init__(self, sensor, request_bridge_update, bridge):
+        self.sensor = sensor
+        self.async_request_bridge_update = request_bridge_update
+        self.bridge = bridge
+        self._last_update_time = self.sensor.lastupdated
+
+    @property
+    def name(self):
+        """Return the name of the ZGP (Hue Tap) Switch."""
+        return self.sensor.name
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the ZGP (Hue Tap) Switch."""
+        return self.sensor.uniqueid
+
+    @property
+    def state(self):
+        """Return the switch's buttonevent if the lastupdated variable changed."""
+        current_time = self.sensor.lastupdated
+        if current_time == self._last_update_time:
+            return None
+        else:
+            self._last_update_time = current_time
+            if self.sensor.buttonevent == ZGP_SWITCH_BUTTON_1:
```

Comment:
If this is a stateless sensor, it's better implemented as home assistant events. See the `deconz` component for example.